src/client.rs

```diff
@@ -1244,6 +1244,7 @@ impl Client {
             .apply_auth(image, RegistryOperation::Push)
             .await?
             .into_request_builder()
+            .header("Content-Length", 0)
```

Comment:
Don't we have to set the actual length? Can you also add an inline comment explaining why we are doing that?

Reply:
Good question, in this case because we are starting the monolithic push _session_ than doing the push itself ([which called after this method is called](https://github.com/oras-project/rust-oci-client/blob/main/src/client.rs#L583)), the spec does not indicate that the `Content-Length` here needs to match the size of the artifact that is being pushed (as you'll see under the [`POST then PUT`](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#post-then-put) section of the OCI distribution spec).

In the case of a [Single `POST`](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#single-post) (which this is not), that would be required, but not when you're obtaining a monolithic push session id.

You'll note that this basically aligns the monolithical push session with that of the [Pushing a blob in chunks](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#pushing-a-blob-in-chunks) section guidance on obtaining a session id.

I realize with this change `begin_push_monolithical_session` is effectively the same as [`begin_push_chunked_session`](https://github.com/oras-project/rust-oci-client/blob/main/src/client.rs#L1258), so I could definitely see the case for consolidating them into a single `begin_push_session` method instead of keeping the two separate methods around, but I was trying to keep the changes minimal and focused on the monolithic pushes alone. Happy to revisit that if you think that would be the better path to go down

Reply:
Thanks for the detailed answer. I appreciate the efforts made to keep the changes minimal